### PR TITLE
@grafana/ui: RadioButtonGroup documentation

### PR DESCRIPTION
**Pull request by jessover9000**
This is a docs update. Added info about how RadioButtonGroups generally work and when to use another component instead.

### CODE REVIEW

File: packages/grafana-ui/src/components/Forms/RadioButtonGroup/RadioButtonGroup.mdx [hunk 1]
```diff
@@ -5,9 +5,13 @@ import { RadioButtonGroup } from './RadioButtonGroup';
 
 # RadioButtonGroup
 
-`RadioButtonGroup` is used for selecting single value from multiple options.
+`RadioButtonGroup` is used to select a single value from multiple mutually exclusive options.
 
-Use `RadioButtonGroup` if there are up to four options available. Otherwise use Select component.
+### When to use
+
+Use `RadioButtonGroup` for mutually exclusive selections if there are up to four options available. For a mutually exclusive selection of more than four options, use `Select` component.
```
Comment (sarlinska) [inline packages/grafana-ui/src/components/Forms/RadioButtonGroup/RadioButtonGroup.mdx]: Why have we decided it's "up to four options"? I am not saying it should be less or more just trying to understand the reasoning.

File: packages/grafana-ui/src/components/Forms/RadioButtonGroup/RadioButtonGroup.mdx [hunk 2]
```diff
@@ -5,9 +5,13 @@ import { RadioButtonGroup } from './RadioButtonGroup';
...
 
-`RadioButtonGroup` is used for selecting single value from multiple options.
+`RadioButtonGroup` is used to select a single value from multiple mutually exclusive options.
 
-Use `RadioButtonGroup` if there are up to four options available. Otherwise use Select component.
+### When to use
+
+Use `RadioButtonGroup` for mutually exclusive selections if there are up to four options available. For a mutually exclusive selection of more than four options, use `Select` component. 
+
+Like `Checkbox`, `RadioButtonGroup` only triggers changes on sending a form. Furthermore, radio buttons can only exist in this group. If you want one single option, or if you want your component to trigger changes immediately, it's better to use `Switch` instead. To offer multiple choices within the same group or context which are not mutually exclusive, use `Checkbox` instead.
```
Comment (sarlinska) [inline packages/grafana-ui/src/components/Forms/RadioButtonGroup/RadioButtonGroup.mdx]: I am not sure "triggering changes on sending a form" will be true for example in Panel Edit. @torkelo ?

File: packages/grafana-ui/src/components/Forms/RadioButtonGroup/RadioButtonGroup.mdx (hunk 1 above)
Comment (tskarhed) [reply packages/grafana-ui/src/components/Forms/RadioButtonGroup/RadioButtonGroup.mdx:12]: I think the reasoning is a matter of both width of the RadioButtonGroup and avoiding clutter.

File: packages/grafana-ui/src/components/Forms/RadioButtonGroup/RadioButtonGroup.mdx (hunk 2 above)
Comment (tskarhed) [reply packages/grafana-ui/src/components/Forms/RadioButtonGroup/RadioButtonGroup.mdx:14]: I agree that this isn't something form exclusive

**Review by tskarhed — APPROVED** [review]

**Pull request merged**